src/main/java/org/reactome/release/dataexport/FTPFileUploader.java

```diff
@@ -0,0 +1,449 @@
+package org.reactome.release.dataexport;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Class for updating Reactome export files on an FTP Server.  Classes which extend this abstract class will provide
+ * methods to:
+ *
+ * 1) Upload new (i.e. current Reactome release) files, created by classes in this project
+ * 2) Delete old (i.e previous Reactome release) files
+ * 3) List files present on the FTP Server for confirmation of successful file upload/deletion.
+ *
+ * @author jweiser
+ */
+public abstract class FTPFileUploader {
+	private static final Logger logger = LogManager.getLogger("mainLog");
+
+	private static final List<String> requiredProperties = Arrays.asList("outputDir", "reactomeVersion");
+
+	private Properties props;
+	private FTPClient ftpClientConnectionToServer;
+
+	/**
+	 * Creates an object for uploading new Reactome files (as well ad deleting old Reactome files) to an FTP Server.
+	 *
+	 * @param props The properties object with the key/value pairs providing connection and path information for
+	 * uploading to the FTP Server
+	 */
+	protected FTPFileUploader(Properties props) {
+		this.props = props;
+		throwIllegalStateExceptionUnlessRequiredPropsPresent();
+	}
+
+	/**
+	 * Returns the logger object for logging info, warnings, errors, etc. from activity in this class and sub-classes
+	 * methods.
+	 *
+	 * @return Logger object for logging class activity
+	 */
+	protected Logger getLogger() {
+		return logger;
+	}
```

Comment:
Why is this method here?

Reply:
I usually prefer to control access to class-level variables, even inside the class itself, so the variable is encapsulated and isn't directly modfiable which I've seen lead to confusing code.  In this case, though, I think you're right and will use the logger variable directly.

Reply:
Done: https://github.com/reactome/release-data-exporter/pull/4/commits/23f8eba7e0885eeaa96062447cc5a181cedadf72